lib/oauth2/request.ex

```diff
@@ -33,14 +46,15 @@ defmodule OAuth2.Request do
       """)
     end
 
-    case :hackney.request(method, url, headers, body, req_opts) do
-      {:ok, ref} when is_reference(ref) ->
-        {:ok, ref}
-
-      {:ok, status, headers, ref} when is_reference(ref) ->
-        process_body(client, status, headers, ref)
-
-      {:ok, status, headers, body} when is_binary(body) ->
+    case Tesla.request(http_client(http_opts),
+           method: method,
+           url: url,
+           query: Keyword.new(params),
```

Comment:
Does it have to be a keyword list?

Reply:
Yeah Map.to_list is better since they also support binary keys:
https://github.com/elixir-tesla/tesla/blob/master/lib/tesla.ex#L33